pypsa_to_IPCC.py

```diff
@@ -121,136 +121,226 @@
+               # MW to Gw
+               if v_type == 'Capacity': 
+                 var[v_type+'|Heating|Heat pumps'] = 0.001*((n.links_t.efficiency.filter(like ='heat pump').filter(like =country).filter(like=str(dict_var[v_type])).mean()
+                    *n.links.p_nom_opt.filter(like ='heat pump').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+                 var[v_type+'|Heating|Electric boilers'] = 0.001*((n.links.efficiency.filter(like ='resistive heater').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='resistive heater').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+               
+               """
+               Capacity : Hydrogen and fuel production
+               """
+               if v_type == 'Capacity':
+                var[v_type+'|Hydrogen|Electricity'] =  0.001*((n.links.efficiency.filter(like ='H2 Electrolysis').filter(like =country).filter(like=str(dict_var[v_type]))
+                  *n.links.p_nom_opt.filter(like ='H2 Electrolysis').filter(like =country).filter(like=str(dict_var[v_type]))).sum())       
+                var[v_type+'|Hydrogen|Gas|w/ CCS'] =  0.001*((n.links.efficiency.filter(like ='SMR CC').filter(like =country).filter(like=str(dict_var[v_type]))
+                  *n.links.p_nom_opt.filter(like ='SMR CC').filter(like =country).filter(like=str(dict_var[v_type]))).sum())                                                                       
+                var[v_type+'|Hydrogen|Gas'] =  0.001*((n.links.efficiency.filter(like ='SMR').filter(like =country).filter(like=str(dict_var[v_type]))
+                  *n.links.p_nom_opt.filter(like ='SMR').filter(like =country).filter(like=str(dict_var[v_type]))).sum()) 
+                var[v_type+'|Hydrogen|Gas|w/o CCS'] = var[v_type+'|Hydrogen|Gas'] -var[v_type+'|Hydrogen|Gas|w/ CCS'] 
+                var[v_type+'|Liquids|Gas'] = 0.001*((n.links.efficiency.filter(like ='Fischer-Tropsch').filter(like =country).filter(like=str(dict_var[v_type]))
+                  *n.links.p_nom_opt.filter(like ='Fischer-Tropsch').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+                var[v_type+'|Gas|Synthetic'] = 0.001*((n.links.efficiency.filter(like ='Sabatier').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='Sabatier').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
             """
-            Electricity : Solar PV, onshore and offshore wind
+            Electricity production : Solar PV, onshore and offshore wind
             """
             #MWh -> EJ
-            var['Secondary Energy|Electricity|Solar|PV'] = 3.6e-9*n.generators_t.p[country + ' solar'].sum() if country + ' solar' in n.generators.index else 0
+            var['Secondary Energy|Electricity|Solar|PV'] = MWh2EJ*h*(n.generators_t.p.filter(like ='solar').filter(like =country).sum().sum())
```

Comment:
where is the variable h defined?

Reply:
I added a definition that uses the scenario name from Config, but could be improved.

Reply:
perfect, I missed that definition